### PR TITLE
Allow flex justification controls to be disabled at the block level

### PR DESCRIPTION
**Pull request by tellthemachines**
## What?

Fixes an issue noticed while reviewing #67022.

The `allowJustification` property of layout support should, per [the docs](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#layout-allowjustification) allow disabling justification controls for the block in constrained or flex layouts, but currently flex justification controls are being rendered regardless of that property.

This PR updates the controls so they are only output when `allowJustification` is `true`.

## Testing Instructions

1. Either create a custom block with layout for testing, or update the layout property of one of the core blocks in its block.json. E.g. in `buttons/block.json` update the layout support to:

```
"layout": {
			"allowSwitching": false,
			"allowInheriting": false,
                         "allowJustification": false,
			"default": {
				"type": "flex"
			}
		},
```

2. Check the block in the editor. The justification controls shouldn't show in the toolbar or in the sidebar.

Note that if orientation controls are enabled (which they are by default), the sidebar will look a bit empty:

Screenshot 2024-11-18 at 4 46 34 pm

But that's already the case when justification is enabled and orientation is disabled (which has been a working feature for a while now):

Screenshot 2024-11-18 at 4 31 06 pm

### CODE REVIEW

File: packages/block-editor/src/layouts/flex.js [hunk 1]
```diff
@@ -66,16 +66,19 @@ export default {
 		onChange,
 		layoutBlockSupport = {},
 	} ) {
-		const { allowOrientation = true } = layoutBlockSupport;
+		const { allowOrientation = true, allowJustification = true } =
```
Comment (ramonjd) [inline packages/block-editor/src/layouts/flex.js:69]: Just double checking the PR's description:

> This PR updates the controls so they are only output when allowJustification is true.

Is it right that `true` is the default?

**Review by ramonjd — APPROVED** [review]
Code looks good to me, and setting `"allowJustification": false` in block.json toggles the controls in the toolbar and in the block settings sidebar


When settings `"allowJustification": false` in block.json:

| Trunk | This PR |
|--------|--------|
| Screenshot 2024-11-18 at 5 45 52 pm | Screenshot 2024-11-18 at 5 45 45 pm |

> the sidebar will look a bit empty:

Better working than empty looking for now is my 2c!

File: packages/block-editor/src/layouts/flex.js [hunk 2]
```diff
@@ -66,16 +66,19 @@ export default {
...
+					{ allowJustification && (
+						<FlexItem>
+							<FlexLayoutJustifyContentControl
+								layout={ layout }
+								onChange={ onChange }
+							/>
+						</FlexItem>
+					) }
 					<FlexItem>
 						{ allowOrientation && (
```
Comment (ntsekouras) [inline packages/block-editor/src/layouts/flex.js]: Can we not render `FlexItem` if `! allowOrientation`? It was from before, but we can update it here.

File: packages/block-editor/src/layouts/flex.js (hunk 1 above)
Comment (tellthemachines) [reply packages/block-editor/src/layouts/flex.js:69]: > Is it right that true is the default?

Yes!

File: packages/block-editor/src/layouts/flex.js (hunk 2 above)
Comment (tellthemachines) [reply packages/block-editor/src/layouts/flex.js]: Yes, updated!

**Pull request merged**